### PR TITLE
add travis CI platform

### PR DESCRIPTION
**Pull request by paulfantom**
Run basic tests in travis-ci platform. This runs same tests as Jenkins instance in "basic" mode, adds 3 types of linters (terraform, yaml, go), and builds package.
This doesn't deploy anything to AWS so it is meant to be run on every PR.

Linters:
- `tflint` is used to lint terraform files in whole repository
- `yamllint` is used to lint every file in `examples` and `installer` directories and subdirectories. It is permissive about line length.
- `golint` recursively checks go files in `installer` directory. For now it is set to be in permissive mode (enforcing version is commented-out) due to some errors in [libvirt.go](installer/pkg/config/libvirt/libvirt.go) file.

### CODE REVIEW

**Comment by paulfantom** [comment]
Travis is already enabled. Working instance is [here](https://travis-ci.org/coreos/tectonic-installer/builds/389227796)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+env:
+  - BAZEL_IMG="quay.io/coreos/tectonic-builder:bazel-v0.3"
+branches:
+  only:
+    - master
+jobs:
+  include:
+    - stage: Lint & Test
+      # Terraform lint
+      script: docker run -v $(pwd):/data -t wata727/tflint
```
Comment (enxebre) [inline .travis.yml]: we'll need to mirror this docker image

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      # YAML lint
+    - script: >
+        docker run -t -v $(pwd):/workdir 
+        giantswarm/yamllint --config-data
```
Comment (enxebre) [inline .travis.yml]: we'll need to mirror this docker image

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      script: docker run -v $(pwd):/data -t wata727/tflint
+      # YAML lint
+    - script: >
+        docker run -t -v $(pwd):/workdir 
+        giantswarm/yamllint --config-data 
+        '{extends: default, rules: {line-length: {level: warning, max: 120}}}' 
+        ./examples/ ./installer/
+      # Go lint
+      #- script: "docker run -v $PWD:$PWD -w $PWD appleboy/golang-testing golint -set_exit_status installer/..."
+    - script: "docker run -v $PWD:$PWD -w $PWD appleboy/golang-testing golint installer/..."
```
Comment (enxebre) [inline .travis.yml]: we'll need to mirror this docker image.
FYI at some point you might wanna try to revive https://github.com/coreos/tectonic-installer/pull/3115 and integrate gometalinter here somehow

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      # Terraform tests
+    - script: "chmod 0777 $PWD && docker run -v $PWD:$PWD:rw -w $PWD $BAZEL_IMG bazel test terraform_fmt --test_output=all"
+      # Installer unit tests
+    - script: "chmod 0777 $PWD && docker run -v $PWD:$PWD:rw -w $PWD $BAZEL_IMG bazel test installer:cli_units --test_output=all"
+    - stage: Build
+      script: "chmod 0777 $PWD && docker run -v $PWD:$PWD:rw -w $PWD $BAZEL_IMG bazel --output_base=.cache build tarball"
```
Comment (enxebre) [inline .travis.yml]: do we need `--output_base=.cache`? we already have a .bazelrc file

File: .travis.yml (hunk 2 above)
Comment (paulfantom) [reply .travis.yml]: And maintain it?

File: .travis.yml (hunk 3 above)
Comment (paulfantom) [reply .travis.yml]: And maintain it?

File: .travis.yml (hunk 4 above)
Comment (paulfantom) [reply .travis.yml]: I wanted to stay close to commands we describe in README.md, but I can remove `--output_base=.cache`

File: .travis.yml (hunk 1 above)
Comment (squat) [reply .travis.yml]: @paulfantom. Yes as a rule we try not to depend on random third party containers whose maintenance and security we cannot guarantee.

Comment (squat) [reply .travis.yml]: @paulfantom. Yes as a rule we try not to depend on random third party containers whose maintenance and security we cannot guarantee.

Comment (paulfantom) [reply .travis.yml]: @enxebre @squat I agree that golang-testing and yamllint containers should be mirrored, but this one (wata727/tflint) is an [official one](https://github.com/wata727/tflint), do we also mirror official ones?

**Pull request merged**